glib/gutils.c

```diff
@@ -3025,7 +3025,24 @@ g_check_setuid (void)
   value = getauxval (AT_SECURE);
   errsv = errno;
   if (errsv)
-    g_error ("getauxval () failed: %s", g_strerror (errsv));
+  {
+    /* When running 32-bit user applications on a 64-bit kernel, reading of the
+     * auxilliary vector can unreliable, likely as a result of the vector not
+     * being architecture agnostic.
+     *
+     * Whilst qemu-user appears to correct these structures depending on the
+     * target architecture, the glibc based loader for armhf (ld-2.27.so) used
+     * by the default toolchain included in the package respositories on Ubuntu
+     * 18.04 does not appear to do so (presumably as the same binary is used on
+     * both 32-bit and 64-bit kernels).
+     *
+     * Since an error results in everything stopping, we instead downgrade the
+     * message to a warning, but return TRUE (indicating that the application
+     * was setuid). Hence we assume the worst case scenario.
+     */
+    g_warning ("getauxval () failed: %s", g_strerror (errsv));
+    return TRUE;
```

Comment:
```suggestion
    return FALSE;
```

Seems like a better failure mode would be `FALSE` here.